lago/utils.py

```diff
@@ -533,12 +534,12 @@ def load_virt_stream(virt_fd):
         dict: Loaded virt config
     """
     try:
-        virt_conf = json.load(virt_fd)
+        virt_conf = json.load(virt_fd, object_pairs_hook=collections.OrderedDict)
     except ValueError:
         virt_fd.seek(0)
         virt_conf = yaml.load(virt_fd)
 
-    return deepcopy(virt_conf)
+    return copy.deepcopy(virt_conf)
```

Comment:
if isinstance(original_obj, list):
        return list(deepcopy(item) for item in original_obj)
    elif isinstance(original_obj, dict):
        return dict((key, deepcopy(val)) for key, val in
original_obj.items())
    elif isinstance(original_obj, collections.OrderedDict):
        return collections.OrderedDict((key, deepcopy(val)) for key, val in
original_obj.items())
    else:
        return original_obj


On Tue, May 24, 2016 at 2:49 PM, David Caro <notifications@github.com>
wrote:

> In lago/utils.py
> <https://github.com/lago-project/lago/pull/262#discussion_r64376092>:
>
> >      except ValueError:
> >          virt_fd.seek(0)
> >          virt_conf = yaml.load(virt_fd)
> >
> > -    return deepcopy(virt_conf)
> > +    return copy.deepcopy(virt_conf)
>
> How did you do that? can you update this pull with that change? (or sent
> another one, or a pastebin)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lago-project/lago/pull/262/files/9e9c7b1a9a17511353aeca189f7ebe75ee61ed4a#r64376092>
>